src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -458,20 +458,41 @@ public static TBuilder UseHelpBuilder<TBuilder>(this TBuilder builder, Func<Bind
         public static CommandLineBuilder UseVersionOption(
             this CommandLineBuilder builder)
         {
-            if (builder.Command.Children.GetByAlias("--version") != null)
+            var command = builder.Command;
+
+            if (command.Children.GetByAlias("--version") != null)
             {
                 return builder;
             }
 
-            var versionOption = new Option("--version", "Show version information");
+            var versionOption = new Option<bool>(
+                "--version",
+                description: "Show version information",
+                parseArgument: result =>
+                {
+                    if (result.FindResultFor(command)?.Children.Count > 1)
```

Comment:
Does the --help also produce an error if combined with other options?
Not really opposed to this behavior, but I do think the --version and --help behavior should match.

Reply:
`--help` is a little different in that it's valid for every single command, whereas `-version` isn't. So:

```console
> command subcommand --version #invalid
> command subcommand --help #valid
```

But for the cases where there additional tokens after `--help` we should maybe consider a similar error.